server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -623,11 +624,42 @@ public boolean configure(final String name, final Map<String, Object> params) th
 
         _allowSubdomainNetworkAccess = Boolean.valueOf(_configs.get(Config.SubDomainNetworkAccess.key()));
 
+        verifyConfigDriveEntriesOnZones();
+
         s_logger.info("Network Service is configured.");
 
         return true;
     }
 
+    /**
+     * Verifies ConfigDrive entries on a zone. Adds Diabled ConfigDrive provider if missing
+     */
+    private void checkConfigDriveEntriesOnZone(DataCenterVO zone) {
+        if (zone.getNetworkType() == NetworkType.Advanced) {
+            List<PhysicalNetworkVO> physicalNetworks = _physicalNetworkDao.listByZoneAndTrafficType(
+                    zone.getId(), TrafficType.Guest);
+            for (PhysicalNetworkVO physicalNetworkVO : physicalNetworks) {
+                PhysicalNetworkServiceProviderVO provider = _pNSPDao.findByServiceProvider(
+                        physicalNetworkVO.getId(), Provider.ConfigDrive.getName());
+                if (provider == null) {
+                    addConfigDriveToPhysicalNetwork(physicalNetworkVO.getId());
+                }
+            }
+        }
+    }
+
+    /**
+     * Verifies ConfigDrive entries on enabled zones. Adds Diabled ConfigDrive provider if missing.
+     */
+    private void verifyConfigDriveEntriesOnZones() {
```

Comment:
What do you think of changing the name from `verifyConfigDriveEntriesOnZones` to `verifyDisabledConfigDriveEntriesOnZones`?

Reply:
What about some test cases here as well?

Reply:
Added tests and changed method names, thanks